cli/resendfailed.php

```diff
@@ -0,0 +1,173 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This script moves records from logstore_xapi_failed_log to logstore_xapi_log.
+ *
+ * @package   logstore_xapi
+ * @author     Scott Verbeek <scottverbeek@catalyst-au.net>
+ * @copyright  2023 Catalyst IT
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define('CLI_SCRIPT', 1);
+require_once(__DIR__ . '../../../../../../../config.php');
+require_once($CFG->libdir . '/clilib.php');
+require_once($CFG->dirroot . '/admin/tool/log/store/xapi/lib.php');
+
+$usage = "Resend logs to the {logstore_xapi_log} table from the {logstore_xapi_failed_log} table.
+
+Usage:
+    # php resendfailed.php --errortype=401,403
+    # php resendfailed.php --datefrom=1698982800 --dateto=1698982823
+    # php resendfailed.php --eventname='\\core\event\\user_loggedin,\\core\\event\\user_loggedout'
+    # php resendfailed.php --datefrom=1698982800 --batch=500 --eventname='\\core\event\\course_viewed' --dryrun=0
+    # php resendfailed.php [--help|-h]
+
+Options:
+    --errortype=<integer(s)>  Comma seperated error types (integer) to resend, default: all.
+    --eventname=<string>      Comma seperated event names to resend, default: all.
+    --datefrom=<integer>      Epoch from date to resend, default: null.
+    --dateto=<integer>        Epoch to date to resend, default: null.
+    --batch=<integer>         The batch size of each move iteration, default: 12500.
+    --dryrun=<integer>        Runs the program without executing any write queries, default: 1.
+    -h --help                 Print this help.
+";
+
+list($options, $unrecognised) = cli_get_params([
+    'help' => false,
+    'errortype' => null,
+    'eventname' => null,
+    'datefrom' => null,
+    'dateto' => null,
+    'dryrun' => true,
+    'batch' => 12500,
+], [
+    'h' => 'help'
+]);
+
+if ($unrecognised) {
+    $unrecognised = implode(PHP_EOL . '  ', $unrecognised);
+    cli_error(get_string('cliunknowoption', 'core_admin', $unrecognised));
+}
+
+if ($options['help']) {
+    cli_writeln($usage);
+    exit(2);
+}
+
+$basetable = XAPI_REPORT_SOURCE_FAILED;
+$extraselect = 'x.errortype, x.response';
```

Comment:
Hey @ScottVerbeek you've got this variable here which isn't used, not sure if that's just leftover from revisions or you forgot to use it?

Reply:
Good catch, this is a remnant of a revision. I'll remove.